test/nn/test_parametrization.py

```diff
@@ -1770,7 +1772,8 @@ def _check_parametrization(
                 type_before_registration == TwoTensor
                 and type_after_registration == Tensor
             ):
-                model._apply(lambda t: TwoTensor(t, t))
+                # TODO: is this requires_grad needed
+                model._apply(lambda t: TwoTensor(t, t, requires_grad=t.requires_grad))
```

Comment:
yes, because you're transforming the parameters and the new parameters want to also require grad consistently as before

Reply:
Oh apparently _apply preserves requires_grad (for the tensor subclass path of _apply) it first wraps the transformed tensor in a parameter with the original requires_grad before swapping:
```
param_applied = torch.nn.Parameter(
    param_applied, requires_grad=param.requires_grad
)
torch.utils.swap_tensors(param, param_applied)
```